src/hotspot/share/runtime/thread.hpp

```diff
@@ -657,15 +657,17 @@ class Thread: public ThreadShadow {
 class ThreadInAsgct {
  private:
   Thread* _thread;
+  bool _saved_in_asgct;
  public:
   ThreadInAsgct(Thread* thread) : _thread(thread) {
     assert(thread != nullptr, "invariant");
-    assert(!thread->in_asgct(), "invariant");
+    // AsyncGetCallTrace is reentrant - save the previous state.
```

Comment:
Nit: It is possible to rephrase this comment as follow:
```
  // Allow AsyncGetCallTrace to be reentrant - save the previous state.
``

Reply:
Thank you for the review - I updated the comment.